R/run-nmtran.R

```diff
@@ -0,0 +1,375 @@
+  status_val <- nmtran.p$status
+  if(is.na(status_val)){
+    cli::cli_abort("NM-TRAN terminated unexpectedly")
+  }else if(status_val == 0){
+    status <- "Finished Running"
+  }else{
+    status <- "Failed. See errors."
+  }
+
+  # Tabulate NM-TRAN results
+  nmtran_results <- list(
+    nmtran_model = basename(nmtran_input),
+    run_dir = run_dir,
+    status = status,
+    status_val = status_val,
+    output = nmtran.p$stdout
+  )
+
+  return(nmtran_results)
+}
+
+
+#' Helper for finding and reading in a `bbi.yaml` file given a `bbi_nonmem_model`
+#' object or explicit `.config_path`.
+#' @inheritParams run_nmtran
+#' @noRd
+get_bbi_yaml_path <- function(.mod = NULL, .config_path = NULL){
+  if(!is.null(.mod)){
+    check_model_object(.mod, "bbi_nonmem_model")
+    model_dir <- get_model_working_directory(.mod)
+  }
+  bbi_yaml_path <- .config_path %||% file.path(model_dir, "bbi.yaml")
+
+  if(!file_exists(bbi_yaml_path)){
+    if(is.null(.config_path)){
+      cli::cli_abort(
+        c(
+          "No bbi configuration was found in the execution directory.",
+          "i" = "Please run {.fun bbr::bbi_init} with the appropriate directory to continue."
+        )
+      )
+    }else{
+      cli::cli_abort("No bbi configuration was found at {.path {(.config_path)}}")
+    }
+  }
+
+  return(bbi_yaml_path)
+}
+
+
+#' Parse `.bbi_args` and return the three expected `nmfe_options` for `NM-TRAN`
+#'  in the correct format
+#' @inheritParams run_nmtran
+#' @param nmfe_options named list of nmfe options defined in a `bbi.yaml` file.
+#'
+#' @details
+#'
+#' Combines NONMEM submission args and consolidates to `NMFE` arguments only
+#'  - The arguments of interest are `prdefault`, `tprdefault`, and `maxlim`,
+#'  which impact the evaluation of `NM-TRAN`
+#'  - Priority: `.bbi_args` > model yaml > `bbi.yaml`
+#'
+#' @examples
+#'
+#' \dontrun{
+#'
+#'  bbi_yaml_path <- get_bbi_yaml_path(.mod)
+#'  bbi_yaml <- yaml::read_yaml(bbi_yaml_path)
+#'
+#'  parse_nmtran_args(
+#'    .mod,
+#'    .bbi_args = list(nm_version = "nm74gf", prdefault = TRUE),
+#'    nmfe_options = bbi_yaml$nmfe_options
+#'  )
+#'
+#' }
+#' @keywords internal
+#' @return a named character vector
+parse_nmtran_args <- function(
+    .mod,
+    .bbi_args = NULL,
+    nmfe_options = NULL
+){
+
+  # These are the default NMFE options that are passed to `NM-TRAN`
+  #  - They should always be passed (in the correct order), so set the
+  #    defaults here and merge with other bbi_args to ensure these are passed
+  #    to NM-TRAN
+  nmfe_args_def <- list(prdefault = FALSE, tprdefault = FALSE, maxlim = 2)
+
+  # Mimic submit_model behavior: a FALSE value for .bbi_arg has no effect
+  # - Filter these out from .bbi_args
+  if(!is.null(.bbi_args)){
+    check_bbi_args(.bbi_args)
+    req_nmtran_args <- paste(names(nmfe_args_def), collapse = "|")
+    .bbi_args <- purrr::imap(.bbi_args, function(val, bbi_arg){
+      if(str_detect(bbi_arg, req_nmtran_args) && isFALSE(val)){
```

Comment:
I don't think it leads to any bugs, but I think constructing a regex here is likely to confuse readers. You're putting together all the options and doing an un-anchored match. That's immediately going to make a reader worry about partial matches. Even among this set, you have "prdefault" matching "tprdefault":

```R
str_detect("tprdefault", "prdefault|tprdefault|maxlim")
#> [1] TRUE
str_detect("tprdefault", "prdefault|maxlim")
#> [1] TRUE
```

It seems likely that we could just get by with reducing the condition to `isFALSE(val)`.

If we need to limit this to the nmfe options, why not just check whether the name is in the vector of names?

Reply:
That's a good point - I dont see any reason why we need to check the names at all, which definitely simplifies this portion